src/borg/helpers/fs.py

```diff
@@ -41,21 +64,17 @@ def get_keys_dir():
     """Determine where to repository keys and cache"""
 
     keys_dir = os.environ.get('BORG_KEYS_DIR', os.path.join(get_config_dir(), 'keys'))
-    if not os.path.exists(keys_dir):
-        os.makedirs(keys_dir)
-        os.chmod(keys_dir, stat.S_IRWXU)
-    return keys_dir
+    if ensure_dir(keys_dir):
+        return keys_dir
```

Comment:
you don't need the "if". just call ensure_dir and return the keys_dir.

it it does not work, it will blow up with some more or less pretty exception (and by default it is the more pretty one).

same applies to all other calls.